src/gateways/translators/BeaconResponseTranslator.ts

```diff
@@ -0,0 +1,91 @@
+import { IBeacon } from "../../entities/IBeacon";
+import { IEmergencyContact } from "../../entities/IEmergencyContact";
+import { IOwner } from "../../entities/IOwner";
+import { IUse } from "../../entities/IUse";
+import { IBeaconResponse } from "./IBeaconResponse";
+
+export interface IBeaconResponseTranslator {
+  translate: (beaconApiResponse: IBeaconResponse) => IBeacon;
+}
+
+export class BeaconResponseTranslator implements IBeaconResponseTranslator {
+  public translate(beaconApiResponse: IBeaconResponse): IBeacon {
+    return {
+      id: beaconApiResponse.data.id,
+      hexId: beaconApiResponse.data.attributes.hexId,
+      type: beaconApiResponse.data.attributes.type,
+      manufacturer: beaconApiResponse.data.attributes.manufacturer,
+      model: beaconApiResponse.data.attributes.model,
+      status: beaconApiResponse.data.attributes.status,
+      registeredDate: beaconApiResponse.data.attributes.createdDate,
+      batteryExpiryDate: beaconApiResponse.data.attributes.batteryExpiryDate,
+      chkCode: beaconApiResponse.data.attributes.chkCode,
+      protocolCode: beaconApiResponse.data.attributes.protocolCode,
+      lastServicedDate: beaconApiResponse.data.attributes.lastServicedDate,
+      manufacturerSerialNumber:
+        beaconApiResponse.data.attributes.manufacturerSerialNumber,
+      owners: this.translateOwners(beaconApiResponse),
+      emergencyContacts: this.translateEmergencyContacts(beaconApiResponse),
+      uses: this.translateUses(beaconApiResponse),
+    };
+  }
+
+  private translateOwners(beaconApiResponse: IBeaconResponse): IOwner[] {
+    const ownerId = beaconApiResponse.data.relationships.owner.data.id;
+
+    return beaconApiResponse.included
+      .filter(
+        (entity) => entity.type === "beaconPerson" && entity.id === ownerId
+      )
+      .map((owner) => {
+        return {
+          id: owner.id,
+          fullName: owner.attributes.fullName,
+          email: owner.attributes.email,
+          telephoneNumber: owner.attributes.telephoneNumber,
+          addressLine1: owner.attributes.addressLine1,
+          addressLine2: owner.attributes.addressLine2,
+          townOrCity: owner.attributes.townOrCity,
+          county: owner.attributes.county,
+          postcode: owner.attributes.postcode,
+        };
+      });
+  }
+
+  private translateEmergencyContacts(
+    beaconApiResponse: IBeaconResponse
+  ): IEmergencyContact[] {
+    const emergencyContactIds = beaconApiResponse.data.relationships.emergencyContacts.data.map(
+      (relationship) => relationship.id
+    );
+
+    return emergencyContactIds.map((emergencyContactId) => {
+      const emergencyContact = beaconApiResponse.included.find(
+        (entity) =>
+          entity.type === "beaconPerson" && entity.id === emergencyContactId
+      )!;
```

Comment:
Consider ETL and how that possibly will not always have emergency contacts

Reply:
@NyWeb @matthew-a-carr The issue here is with there being an emergency contact ID in the `relationships` property of the APIResponse but that emergency contact's details are not found in the `included` property.  I've added a guard clause to throw if that happens, which has made the TS warning go away:

```typescript
  private mapEmergencyContacts(
    beaconApiResponse: IBeaconResponse
  ): IEmergencyContact[] {
    const emergencyContactIds = beaconApiResponse.data.relationships.emergencyContacts.data.map(
      (relationship) => relationship.id
    );

    return emergencyContactIds.map((emergencyContactId) => {
      const emergencyContact = beaconApiResponse.included.find(
        (entity) =>
          entity.type === "beaconPerson" && entity.id === emergencyContactId
      );

      // New code 
      if (!emergencyContact)
        throw ReferenceError(`${emergencyContactId} is defined as a relationship but not found in "included".  This is 
      likely to be a problem with the API response`);
      // New code
      
      return {
        id: emergencyContactId,
        fullName: emergencyContact.attributes.fullName,
        telephoneNumber: emergencyContact.attributes.telephoneNumber,
        alternativeTelephoneNumber:
          emergencyContact.attributes.alternativeTelephoneNumber,
      };
    });
  }
```

This will capture internally inconsistent API responses, which shouldn't be an issue if they're autogenerated.  (I.e., it's very unlikely to be the case that an emergency contact relationship is included in `relationships` but not in `included`, but if it does happen we'll find out with a server error).  

This code still handles instances when a beacon has no emergency contacts at all, e.g. legacy data post-ETL.